foundry.md

```diff
@@ -537,15 +484,106 @@ Utils
-         <callGas> GCALL => 0 </callGas>
+    syntax KItem ::= "#setBalance" Int Int [klabel(foundry_setBalance)]
+ // -------------------------------------------------------------------
+    rule <k> #setBalance ACCTID NEWBAL => . ... </k>
+         <account>
+           <acctID> ACCTID </acctID>
+           <balance> _ => NEWBAL </balance>
+           ...
+         </account>
+```
+
+- `#setCode ACCTID CODE` sets the code of a given account.
+
+```k
+    syntax KItem ::= "#setCode" Int ByteArray [klabel(foundry_setCode)]
+ // -------------------------------------------------------------------
+    rule <k> #setCode ACCTID CODE => . ... </k>
+         <account>
+           <acctID> ACCTID </acctID>
+           <code> _ => #if #asWord(CODE) ==Int 0 #then .ByteArray:AccountCode #else {CODE}:>AccountCode #fi </code>
+           ...
+         </account>
+```
+
+- `#returnNonce ACCTID` takes the nonce of a given account and places it on the `<output>` cell.
+
+```k
+    syntax KItem ::= "#returnNonce" Int [klabel(foundry_returnNonce)]
+ // -----------------------------------------------------------------
+    rule <k> #returnNonce ACCTID => . ... </k>
+         <output> _ => #bufStrict(32, NONCE) </output>
+         <account>
+           <acctID> ACCTID </acctID>
+           <nonce> NONCE </nonce>
+           ...
+         </account>
+```
+
+- `#setNonce ACCTID NONCE` takes a given account and a given nonce and update the account accordingly.
+
+```k
+     syntax KItem ::= "#setNonce" Int Int [klabel(foundry_setNonce)]
+ // ----------------------------------------------------------------
+    rule <k> #setNonce ACCTID NONCE => . ... </k>
+         <account>
+             <acctID> ACCTID </acctID>
+             <nonce> _ => NONCE </nonce>
+             ...
+         </account>
+```
+
+- `#returnStorage ACCTID LOC` takes a storage value from a given account and places it on the `<output>` cell.
+
+```k
+    syntax KItem ::= "#returnStorage" Int Int [klabel(foundry_returnStorage)]
+ // -------------------------------------------------------------------------
+    rule <k> #returnStorage ACCTID LOC => . ... </k>
+         <output> _ => #bufStrict(32, #lookup(STORAGE, LOC)) </output>
+         <account>
+           <acctID> ACCTID </acctID>
+           <storage> STORAGE </storage>
+           ...
+         </account>
 ```
 
+- `#setStorage ACCTID LOC VALUE` sets a given value to a given location of an account.
+
+```k
+    syntax KItem ::= "#setStorage" Int Int Int [klabel(foundry_setStorage)]
+ // -----------------------------------------------------------------------
+    rule <k> #setStorage ACCTID LOC VALUE => . ... </k>
+         <account>
+           <acctID> ACCTID </acctID>
+           <storage> STORAGE => STORAGE [ LOC <- VALUE ] </storage>
+             ...
+         </account>
+```
+
+ - selectors for cheat code functions.
+
+```k
+    syntax Int ::= selector ( String ) [alias, klabel(abi_selector), symbol, function, no-evaluators]
```

Comment:
This should be moved back to where it was defined, because there are other modules which use it.

Reply:
Moved to `abi.md` instead.